### PR TITLE
Add Pedia article for Design Window in User Interface Category.

### PR DESCRIPTION
**Pull request by LGM-Doyle**
This is intended to address #671.

### CODE REVIEW

File: default/stringtables/en.txt [hunk 1]
```diff
@@ -4219,6 +4219,39 @@ Next are icons showing the player empire total [[encyclopedia INDUSTRY_TITLE]],
 
 On the right hand side of the control bar are a number of icons for toggling display of other significant UI windows. Tooltips are available for these.'''
 
+DESIGN_WINDOW_ARTICLE_TITLE
+Design Window
+DESIGN_WINDOW_ARTICLE_TEXT
+'''The design window is for viewing or altering blueprints of monsters or ships.
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: Why introduce "blueprints"? Not used anywhere else / before, is it? Seems like unnecessary extra jargon.

Comment (LGM-Doyle) [inline default/stringtables/en.txt]: I'm trying to create a difference in meaning between blueprints as the plan for a ship, the queued production of a ship and the ship which comes off the end of the production line.  

It is to get ahead of the question, "Why doesn't changing the design change all of the already queued designs as well?".  If people are asking themselves, "Why doesn't changing the blueprint, change the design in the queue?", then I was hoping the answer would be clear.

The question was asked [here.](http://freeorion.org/forum/viewtopic.php?f=25&t=10090&sid=d961978af4ed03a5c22e0d7bcad1a375)  I think that it is a question that will keep arising unless the UI is more obvious in the way it works.

So, I could either remove the word blueprint, or go all in and make the design window the blueprint window?

Comment (geoffthemedio) [inline default/stringtables/en.txt]: Remove the word blueprint.

**Comment by LGM-Doyle** [comment]
@geoffthemedio I force pushed a no blueprint version.

File: default/stringtables/en.txt [hunk 2]
```diff
@@ -4219,6 +4219,39 @@ Next are icons showing the player empire total [[encyclopedia INDUSTRY_TITLE]],
...
+
+
+Design Starts has four tabs: empty Hulls, Finished designs, Saved designs from previous games and Monster designs for inspection.  Drag or double click hulls, designs or monsters to edit/examine them on the workbench.
+
+Finished designs appear in the same order as on the Production Window.  Drag and drop designs within the list to change the order.  Delete designs with Obsolete from the context menu.  Ships already queued for production will continue to be built with the old design.  Save designs to disk for future games with Save from the context menu.
+
+Right click Saved designs and copy a single or all saved designs to the empire's known designs.
+
+
+Ship Parts shows parts that can be added to the current design.
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: Not really true. Parts that can't be added to a design are also shown.

File: default/stringtables/en.txt [hunk 3]
```diff
@@ -4219,6 +4219,39 @@ Next are icons showing the player empire total [[encyclopedia INDUSTRY_TITLE]],
...
+
+The Available and Unavailable filters on Design Starts and Ship Parts toggle the display of parts that the empire can currently use or that are restricted.
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: some note about why parts would be restricted, and how to unrestrict them, would be useful...

File: default/stringtables/en.txt [hunk 4]
```diff
@@ -4219,6 +4219,39 @@ Next are icons showing the player empire total [[encyclopedia INDUSTRY_TITLE]],
...
+
+The Ship Parts window has many filters to display/hide entire categories of parts.
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: nix "many"

File: default/stringtables/en.txt [hunk 5]
```diff
@@ -4219,6 +4219,39 @@ Next are icons showing the player empire total [[encyclopedia INDUSTRY_TITLE]],
...
+
+
+Design Details is the workbench for the current design.
+
+The current design may have internal, external and/or core slots.  Drag and drop or double click parts from Ship Parts to add/remove them from the design.
+
+Add a name and possibly a description to the design.
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: possibly -> optionally

File: default/stringtables/en.txt [hunk 6]
```diff
@@ -4219,6 +4219,39 @@ Next are icons showing the player empire total [[encyclopedia INDUSTRY_TITLE]],
...
+
+Click Add New Design and the design will be added to the end of the list of Finished designs.
+
+Click Update Old Design to replace the most recently selected finished design with a new design.  Update Old Design is only active if this design was started from a design in the Finished list.  Update Old Design only changes the design, any ships already queued for production will continue to use the old design.
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: and existing / already produced ships will be unchanged.

File: default/stringtables/en.txt (hunk 3 above)
Comment (MatGB) [inline default/stringtables/en.txt]: Yes, I also wonder if there should be an 'unresearchable' toggle or other hiding mechanism for the stuff that isn't meant to be on player designs, there's a big difference for a player between "you haven't got this yet" and "you'll never have this"

**Comment by LGM-Doyle** [comment]
@MatGB I agree.

It also concerns me that one of the states when both Available and Unavailable are untoggled shows no parts.  In that state if the player doesn't realize how they ended up with no parts displayed the design window is unusable.  It seems like an unnecessary gotcha.

Here is a thought.
Ship Parts could have 3 expandable/collapsable sections Available, Unresearched and Foreign.  Foreign is for unresearchable items like monster tech, experimenter tech and possibly other empire tech if we eventually do randomized or per species tech trees. 

If all sections are collapsed it would be clearer why there are no parts displayed.

It removes the available/unavailable toggles from the visually identical category buttons.

The unavailable parts wouldn't be interleaved with the available parts.  This may be a good/bad thing, depending on exactly how your are using the interface at the moment.

File: default/stringtables/en.txt [hunk 7]
```diff
@@ -4219,6 +4219,39 @@ Next are icons showing the player empire total [[encyclopedia INDUSTRY_TITLE]],
...
 
+DESIGN_WINDOW_ARTICLE_TITLE
+Design Window
+DESIGN_WINDOW_ARTICLE_TEXT
+'''The design window is for viewing or altering the designs of monsters or ships.
+
+There are four sub windows:  Design Starts, Ship Parts, Design Details and the Pedia.
+
+
+Design Starts has four tabs: empty Hulls, Finished designs, Saved designs from previous games and Monster designs for inspection.  Drag or double click hulls, designs or monsters to edit/examine them on the workbench.
```
Comment (geoffthemedio) [inline default/stringtables/en.txt:4241]: capitalization of Starts of words Seems Random...

File: default/stringtables/en.txt [hunk 8]
```diff
@@ -4219,6 +4219,39 @@ Next are icons showing the player empire total [[encyclopedia INDUSTRY_TITLE]],
...
+
+Finished designs appear in the same order as on the Production Window.  Drag and drop designs within the list to change the order.  Delete designs with Obsolete from the context menu.  Ships already queued for production will continue to be built with the old design.  Save designs to disk for future games with Save from the context menu.
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: oddly mixes imperative and descriptive sentences. should be in separate paragraphs, or precede the instructions with "you can", or better yet, rephrase to "Designs can be drag-and-dropped within the list to change their order."

File: default/stringtables/en.txt [hunk 9]
```diff
@@ -4219,6 +4219,39 @@ Next are icons showing the player empire total [[encyclopedia INDUSTRY_TITLE]],
...
+
+Right click Saved designs and copy a single or all saved designs to the empire's known designs.
+
+
+Ship Parts displays parts. Parts whose associated technology has been researched may be used in current designs.  Other parts can not be used because either the associated technology has not been researched or the part is only available to monsters.
+
+The Available and Unavailable filters on Design Starts and Ship Parts toggle the display of parts that the empire can currently use or that are restricted.  Researching the associated technology makes parts available.  Some parts will never be available because they are only used for monsters.
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: Is that the only reason are part could not be possible to unlock?

File: default/stringtables/en.txt (hunk 7 above)
Comment (LGM-Doyle) [inline default/stringtables/en.txt:4241]: I've capitalized words that are tab names from the UI, or context menu commands.  I could remove the capitalization or use quotes.

File: default/stringtables/en.txt (hunk 9 above)
Comment (LGM-Doyle) [inline default/stringtables/en.txt]: I'm not certain but I think so.  That is assuming monsters includes experimentors.

Do you know otherwise?

Comment (dbenage-cx) [inline default/stringtables/en.txt]: Might drop it and the previous sentence.  
AFAIK, the only part not available from techs or strictly a monster is Krill Spawner.  Which could be completely unavailable if the player sets games to no specials.  This part might still fit the description, but future additions could come from a variety of conditions (granted with species; single-player only).

File: default/stringtables/en.txt (hunk 7 above)
Comment (geoffthemedio) [inline default/stringtables/en.txt:4241]: That makes sense as it is then; it's fine.

File: default/stringtables/en.txt (hunk 9 above)
Comment (geoffthemedio) [inline default/stringtables/en.txt]: I'd just write something like "Some unavailable parts may be unlocked by researching associated technologies."

**Comment by geoffthemedio** [comment]
Can you squash into one commit?

**Comment by LGM-Doyle** [comment]
Rebased to master and squashed.

**Pull request merged**